timemachine/fe/free_energy.py

```diff
@@ -74,15 +74,21 @@ class PairBarPlots:
     overlap_detail_png: bytes
 
 
+@dataclass
+class IntermediateResult:
+    initial_states: List[InitialState]
+    pair_bar_results: List[PairBarResult]
+
+
 @dataclass
 class SimulationResult:
     initial_states: List[InitialState]
-    result: PairBarResult
+    pair_bar_results: List[PairBarResult]
```

Comment:
This change to make `PairBarResult` represent the result for a single pair of states simplifies the implementation of caching. If we want to avoid downstream changes, we could introduce a new internal type and avoid modifying this.

Reply:
This change make sense to me.
-------
Nit: Noting a small name shift / collision, since current state is a bit confusing:

Current:
* "BAR" -- estimator that accepts 2 states
* "Pair BAR" -- estimator that accepts a list of K states, and applies BAR to the K-1 adjacent pairs
* "MBAR" -- estimator that generalizes BAR to K states

Proposed:
* "Pair BAR" -- synonymous with "BAR"

(not sure if there's a less ambiguous name for what we're currently calling "pair BAR", to avoid collision with "BAR on a single pair"...)

Reply:
I think that makes sense. I think the convention in this PR (i.e., calling the result for a pair of states `PairBarResult`, instead of the latter referring to $L-1$ adjacent pairs) matches the proposed; is that correct?

Reply:
>I think that makes sense. I think the convention in this PR ... matches the proposed; is that correct?

Correct.

(Sorry for unclarity -- not advocating for any change relative to your PR, just observing a name shift between "current" (master) and "proposed" (this PR).)

Reply:
> not sure if there's a less ambiguous name for what we're currently calling "pair BAR", to avoid collision with "BAR on a single pair"...

Ah, I think I might have misunderstood this comment before (I assumed you were proposing the convention "Pair BAR" = "BAR", rather than noting that this seems to be implied by the renaming in this PR).

I'm now leaning toward keeping the current convention of "Pair BAR" referring to the estimator that accepts a list of K states, and applies BAR to the K-1 adjacent pairs, and renaming the type representing the result for a single pair to just `BARResult` or similar.

Reply:
Updated the naming convention in https://github.com/proteneer/timemachine/pull/974/commits/ebcfe5d0f40c9e6460eb7de50548b812d01d874a:

* Rename IntermediateResult -> PairBarResult
* Use "pair BAR" to refer to estimator that accepts a list of K states, and applies BAR to the K-1 adjacent pairs
* Use "BAR" to refer to BAR on a single pair of states